var/spack/repos/builtin/packages/ufs-weather-model/package.py

```diff
@@ -17,68 +17,160 @@ class UfsWeatherModel(CMakePackage):
     url = "https://github.com/ufs-community/ufs-weather-model/archive/refs/tags/ufs-v1.1.0.tar.gz"
     git = "https://github.com/ufs-community/ufs-weather-model.git"
 
-    maintainers("t-brown")
+    maintainers("AlexanderRichert-NOAA")
 
-    version(
-        "2.0.0",
-        tag="ufs-v2.0.0",
-        commit="e3cb92f1cd8941c019ee5ef7da5c9aef67d55cf8",
-        submodules=True,
-    )
-    version(
-        "1.1.0",
-        tag="ufs-v1.1.0",
-        commit="5bea16b6d41d810dc2e45cba0fa3841f45ea7c7a",
-        submodules=True,
-    )
+    version("develop", branch="develop", submodules=True)
+    version("2.0.0", tag="ufs-v2.0.0", submodules=True)
+    version("1.1.0", tag="ufs-v1.1.0", submodules=True)
 
+    variant("mpi", default=True, description="Enable MPI")
     variant(
         "32bit", default=True, description="Enable 32-bit single precision arithmetic in dycore"
     )
-    variant("avx2", default=False, description="Enable AVX2 instructions")
     variant(
-        "ccpp", default=True, description="Enable the Common Community Physics Package (CCPP))"
+        "ccpp_32bit",
+        default=False,
+        description="Enable CCPP_32BIT (single precision arithmetic in slow physics)",
+    )
+    variant("avx2", default=True, description="Enable AVX2 instruction set")
+    variant(
+        "simdmultiarch", default=False, description="Enable multi-target SIMD instruction sets"
+    )
+    variant("debug", default=False, description="Enable DEBUG mode", when="@develop")
+    variant(
+        "debug_linkmpi",
+        default=True,
+        description="Enable linkmpi option when DEBUG mode is on",
+        when="@develop",
+    )
```

Comment:
Is there a reason for these options to only be available on develop?

Reply:
All of the variant with 'when="@develop"' are newer options not found in the older versions.

Reply:
Yes, my point is they aren't specific to the develop branch -- if you want to be forward-thinking, you can make them be `@2.1:` so that there is less work to do when future versions with these features are added.

But if you expect develop to be the only version users care about I guess it doesn't matter.

Reply:
Yeah in practice it'll only be used for develop. I'm not even sure if anyone currently uses the numbered versions here, so I'll probably take those out at some point down the road just to clean things up.

Reply:
I have the same concerns as @becker33 here. Spack defaults to stable checksummable releases. We still want to keep these, and add new versions in the future. Even if you only use develop, other users will likely prefer a stable release so they can reproduce the results of their research. My suggestion would be to use `@2.1:` for things that were added in 2.1, `@:2.0` for things that exist in 2.0.X but don't in 2.1.X, etc. But you're the package maintainer, so technically you get the final say in what you want to support. So if you only care about develop and no one else volunteers to support these stable releases, you are technically allowed to deprecate and remove them.

Reply:
The tricky thing here is that this model doesn't really follow regular, sequential releases. In fact, there hasn't been a "regular" release since 2.0. All the other tags/releases are specific to configurations that we're not trying to support through the Spack recipe (for instance, there are tags for operational configurations that are only used internally to NOAA, and those will not be installed with Spack). Would it be okay to leave it as is, but with the goal of changing these settings to specific versions if/when new versions emerge?

Reply:
I'm fine with leaving it as is for now. Are you a developer of UFS? Maybe you can encourage the other devs to make more regular releases in the future so the versions have some kind of meaning. This is really important for experiment reproducibility.